src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -882,22 +879,19 @@ def generic_wait_command(self, name, getter_name='get', getter_type=None, **kwar
 
         if getter_type:
             merged_kwargs = _merge_kwargs(getter_type.settings, merged_kwargs, CLI_COMMAND_KWARGS)
-        getter_op = self._resolve_operation(merged_kwargs, getter_name, getter_type)
-        _cli_generic_wait_command(
-            self.command_loader,
-            '{} {}'.format(self.group_name, name),
-            getter_op=getter_op,
-            **merged_kwargs)
+        getter_op = self._resolve_operation(merged_kwargs, getter_name, getter_type, custom_type=custom_command)
+        _cli_generic_wait_command(self.command_loader, '{} {}'.format(self.group_name, name), getter_op=getter_op,
+                                  custom_command=custom_command, **merged_kwargs)
 
-    def generic_show_command(self, name, getter_name='get', getter_type=None, **kwargs):
+    def generic_show_command(self, name, getter_name='get', getter_type=None, custom_command=False, **kwargs):
```

Comment:
I don't see the simplification here. Previously if you have a custom getter, you would specify the getter name and `getter_type=custom_type`.  This seems like it opens up the potential to have conflicting arguments.  What happens if I put `getter_type=custom_type` and `custom_command=False`?

Reply:
The `getter_type` and `setter_type` will always take priority.
If `getter_type=custom_type`, then the custom_type will be used.
This should simplify things since most `getter_type`/`setter_type`s are custom.
ex. https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/commands.py#L35-L45

In the example above, 3 lines would be saved for the one generic_update command as there would be no need to redefine the `custom_command_type`.